### PR TITLE
[cmake] addons support multiple regex arguments for ADDONS_TO_BUILD

### PR DESCRIPTION
**Pull request by fuzzard**
## Description
Currently if supplying multiple arguments for ADDONS_TO_BUILD that include a regex
```
ADDONS_TO_BUILD='game.libretro.* peripheral.joystick'
```
The regex will not match or be processed at all.
You currently need to explicitly set each if you include multiple arguments
```
ADDONS_TO_BUILD='game.libretro.2048 game.libretro.mrboom peripheral.joystick'
```

## Motivation and Context
Playing with libretro addons, found that a regex isnt used if supplied in a multi argument addons_to_build

## How Has This Been Tested?
Mac OS building  multiple combinations of selective arguments

```
ADDONS_TO_BUILD='game.libretro.* peripheral.joystick'
```
exact match of game.libretro and peripheral.joystick, regex match of game.libretro.mrboom, game.libretro.2048
```
ADDONS_TO_BUILD='game.libretro peripheral.joystick'
```
exact match of game.libretro and peripheral.joystick, regex match of game.libretro.mrboom, game.libretro.2048
```
ADDONS_TO_BUILD='pvr.hdhomerun pvr.argus* peripheral.joystick'
```
Exact match of pvr.hdhomerun and peripheral.joystick, with regex match of pvr.argustv

This also highlights an existing behaviour regarding the core game.libretro. You cannot selectively only build game.libretro as all other game.libretro.* automatically match with the cmake if check. This behaviour does not change with this PR.

https://github.com/xbmc/xbmc/blob/2d7e977792ff09ea865ac35caeca8fabf98e5523/cmake/addons/CMakeLists.txt#L259

I think the easier way to resolve this would be to rename game.libretro to something like game.libretro.core. I couldnt come up with anything easy to handle this case, and game.libretro is the only case i could find this happens to across any binary addons. No other active binary addon, has the name that is a subset of other active addons

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [X] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Review by phunkyfish — APPROVED** [review]
Nice change. Tested locally, works well.

File: cmake/addons/CMakeLists.txt [hunk 1]
```diff
@@ -254,11 +254,16 @@ foreach(addon ${addons})
     # try to find a perfect match
     list(FIND ADDONS_TO_BUILD ${id} idx)
     if(idx GREATER -1 OR "${ADDONS_TO_BUILD}" STREQUAL "all")
+      message(STATUS "Exact Match ${id}, building addon")
```
Comment (wsnipex) [inline cmake/addons/CMakeLists.txt:257]: minor: this msg is only correct if ADDONS_TO_BUILD != "all"

File: cmake/addons/CMakeLists.txt [hunk 2]
```diff
@@ -254,11 +254,16 @@ foreach(addon ${addons})
...
       set(ADDON_FOUND TRUE)
     # Maybe we have a regex
-    elseif(id MATCHES "${ADDONS_TO_BUILD}")
-      message(STATUS "Pattern ${ADDONS_TO_BUILD} matches ${id}, building addon")
-      set(ADDON_FOUND TRUE)
+    else()
+      foreach(ADDONLISTITEM ${ADDONS_TO_BUILD})
+        if(id MATCHES "${ADDONLISTITEM}")
+          message(STATUS "Pattern ${ADDONLISTITEM} matches ${id}, building addon")
+          set(ADDON_FOUND TRUE)
```
Comment (AlwinEsch) [inline cmake/addons/CMakeLists.txt:264]: Think whould be good to add a `break()` below, as after a `TRUE` the next need no more checked.

File: cmake/addons/CMakeLists.txt (hunk 1 above)
Comment (AlwinEsch) [inline cmake/addons/CMakeLists.txt:257]: Yes, there the name not 100% correct if the `all` is used.

**Review by AlwinEsch — COMMENTED** [review]
Your changes are good to have for such cases :+1:.
Thanks for this.

File: cmake/addons/CMakeLists.txt (hunk 1 above)
Comment (fuzzard) [reply cmake/addons/CMakeLists.txt:257]: I can just drop this change, was just useful to get a message to know if the exact match was hit.

File: cmake/addons/CMakeLists.txt (hunk 2 above)
Comment (fuzzard) [reply cmake/addons/CMakeLists.txt:264]: Will add this, cheers

File: cmake/addons/CMakeLists.txt (hunk 1 above)
Comment (fuzzard) [reply cmake/addons/CMakeLists.txt:257]: As mentioned above, will just drop this change

**Review by wsnipex — APPROVED** [review]
looks good

**Review by AlwinEsch — APPROVED** [review]

**Pull request merged**